test/test_mps.py

```diff
@@ -8514,14 +8514,18 @@ def helper(shape, op, inplace, dtype=torch.float32):
                 op(mps_x, out=mps_y)
                 self.assertEqual(mps_y, cpu_y)
 
-                # test for non contiguous input/output with similar strides
+                # test for non contiguous but dense input/output with similar strides
                 cpu_x = torch.randn(shape, device='cpu', dtype=dtype).mT
                 mps_x = cpu_x.to('mps')
                 cpu_y = torch.empty_like(cpu_x)
                 mps_y = cpu_y.to('mps')
                 op(cpu_x, out=cpu_y)
                 op(mps_x, out=mps_y)
                 self.assertEqual(mps_y, cpu_y)
+                # test for sliced inputs and outputs with similar strides
+                mps_x, mps_y = torch.randn((2, shape[0] * 2, shape[1] * 2), device='mps', dtype=dtype).unbind(0)
+                op(mps_x[::2, ::2], out=mps_y[::2, ::2])
+                self.assertEqual(mps_y[::2, ::2], op(mps_x[::2, ::2].contiguous()))
```

Comment:
this is a little harder to parse for reference compared to using the same call on cpu